### PR TITLE
[connector/spanmetricsv2] Use slice for spans config

### PR DESCRIPTION
**Pull request by lahsivjar**
Allows creating span metrics with same metric name but different resource attributes.

### CODE REVIEW

**Comment by lahsivjar** [comment]
Putting in draft, need to fix the attribute handling.

**Review by carsonip — APPROVED** [review]

**Comment by lahsivjar** [comment]
@carsonip There was an edge case with same name and description which was not handled correctly. I have fixed this but also force-pushed it. PTAL.

File: connector/spanmetricsconnectorv2/config.go [hunk 1]
```diff
@@ -44,17 +44,44 @@ const (
...
+	keyMap := make(map[string]AttributeConfig)
+	for _, attr := range mi.Attributes {
+		keyMap[attr.Key] = attr
+	}
+
+	for _, otherAttr := range other.Attributes {
+		if _, ok := keyMap[otherAttr.Key]; !ok {
+			return false
+		}
+	}
```
Comment (carsonip) [inline connector/spanmetricsconnectorv2/config.go:81]: I see this handles `mi.Attributes` being a subset of `other.Attributes`. But does this handle `other.Attributes` being a strict subset of `mi.Attributes`?

File: connector/spanmetricsconnectorv2/config.go [hunk 2]
```diff
@@ -44,17 +44,44 @@ const (
...
+		return true
+	}
+	// Validate attribues equality
+	keyMap := make(map[string]AttributeConfig)
+	for _, attr := range mi.Attributes {
+		keyMap[attr.Key] = attr
+	}
+
+	for _, otherAttr := range other.Attributes {
+		if _, ok := keyMap[otherAttr.Key]; !ok {
```
[the remark this answers is not in the document]
Comment (lahsivjar) [reply connector/spanmetricsconnectorv2/config.go:78]: Intentionally didn't account for it, I guess we can add it if/when needed.

**Review by carsonip — APPROVED** [review]

**Pull request merged**